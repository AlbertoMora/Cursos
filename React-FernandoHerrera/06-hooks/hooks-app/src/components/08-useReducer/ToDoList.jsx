import React from "react";
import ToDoListItem from "./ToDoListItem";

const ToDoList = ({ toDos, handleDelete, handleToggle }) => {
    return (
        <ul className="list-group list-group-flush">
            {toDos.map((todo, index) => (
                <ToDoListItem
                    key={todo.id}
                    todo={todo}
                    index={index}
                    handleDelete={handleDelete}
                    handleToggle={handleToggle}
                />
            ))}
        </ul>
    );
};

export default ToDoList;
