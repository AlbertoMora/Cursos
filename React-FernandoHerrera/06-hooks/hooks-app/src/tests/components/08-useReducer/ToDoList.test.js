import React from "react";
import { shallow } from "enzyme";
import ToDoList from "../../../components/08-useReducer/ToDoList";

import { demoTodos } from "./fixtures/demoTodos";

describe("Tests on <TodoList />", () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<ToDoList toDos={demoTodos} handleDelete={handleDelete} handleToggle={handleToggle} />);

    test("should render the component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should have demoTodos.length number of elements", () => {
        expect(wrapper.find("ToDoListItem").length).toBe(demoTodos.length);
    });
    test("should ToDoListItem children have the right properties", () => {
        const item = wrapper.find("ToDoListItem").at(0);

        expect(item.prop("handleDelete")).toEqual(expect.any(Function));
        expect(item.prop("handleToggle")).toEqual(expect.any(Function));
        expect(item.prop("todo")).toEqual(expect.any(Object));
        expect(item.prop("index")).toEqual(expect.any(Number));
    });
});
