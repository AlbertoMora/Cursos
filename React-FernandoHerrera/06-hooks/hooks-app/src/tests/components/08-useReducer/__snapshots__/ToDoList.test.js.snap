// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <TodoList /> should render the component correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <ToDoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <ToDoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
